File: javascript/first_unique_character.js
//Given a string, find the first non-repeating character in it and return it's index. If it doesn't exist, return -1.

//Example 1:
//s = "leetcode"
//return 0.

//Example 2:
//s = "loveleetcode",
//return 2.

//Note: You may assume the string contain only lowercase letters.

const firstUniqChar = s => {
    let hash = {}
    for(let i=0; i<s.length; i++) {
        if(s[i] in hash) hash[s[i]][1]++
        else hash[s[i]] = [i,1]
    }
    for(const char in hash)
        if(hash[char][1] == 1) return hash[char][0]
    return -1
}

let ex1 = "leetcode"
let ex2 = "loveleetcode"
let ex3 = "dddccdbba"

console.log(firstUniqChar(ex1))
console.log(firstUniqChar(ex2))
console.log(firstUniqChar(ex3))

File: python/first_unique_character.py
# Given a string, find the first non-repeating character in it and return it's index. If it doesn't exist, return -1.

# Example 1:
# s = "leetcode"
# return 0.

# Example 2:
# s = "loveleetcode",
# return 2.

# Note: You may assume the string contain only lowercase letters.

def firstUniqChar(s):
    hash = {}
    for i in range(len(s)):
        if s[i] in hash: hash[s[i]][1] += 1
        else: hash[s[i]] = [i,1]
    for item in hash:
        if hash[item][1] == 1: return hash[item][0]
    return -1

ex1 = "leetcode"
ex2 = "loveleetcode"
ex3 = "dddccdbba"

print(firstUniqChar(ex1))
print(firstUniqChar(ex2))
print(firstUniqChar(ex3))